Sources/Plasma/PubUtilLib/plCompression/plZlibCompress.cpp

```diff
@@ -246,34 +248,33 @@ bool  plZlibCompress::CompressToFile( hsStream * s, const char * filename )
 {
     gzFile  outFile;
     bool    worked = false;
-    int     length, err;
-
-    uint8_t   buffer[ kGzBufferSize ];
+    int     err;
 
+    auto buffer = std::make_unique<uint8_t[]>(kGzBufferSize);
 
     outFile = gzopen( filename, "wb" );
     if( outFile != nil )
     {
         for( ;; )
         {
-            int avail = s->GetEOF()-s->GetPosition();
-            int n = ( avail>sizeof( buffer ) ) ? sizeof( buffer ) : avail;
+            uint32_t avail = s->GetEOF()-s->GetPosition();
```

Comment:
```suggestion
            uint32_t avail = s->GetEOF() - s->GetPosition();
```
(The subtraction wasn't obvious to me)